### PR TITLE
[14.0][IMP] account: speed up some queries

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Be faster.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Have you measured that this is really an improvement?

**Comment by MiquelRForgeFlow** [comment]
This weekend will check.

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 1]
```diff
@@ -209,6 +209,7 @@ def add_move_id_field_account_bank_statement_line(env):
         SET statement_line_id = aml.statement_line_id
         FROM account_move_line aml
         WHERE aml.move_id = am.id AND aml.statement_line_id IS NOT NULL
+            AND am.statement_line_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:212]: before 0:00:10.667838 running
after    0:00:08.039494 running

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 2]
```diff
@@ -218,7 +219,7 @@ def add_move_id_field_account_bank_statement_line(env):
         UPDATE account_bank_statement_line absl
         SET move_id = am.id
         FROM account_move am
-        WHERE am.statement_line_id = absl.id
+        WHERE am.statement_line_id = absl.id AND absl.move_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:222]: before 0:00:03.573477 running
after 0:00:03.597853 running

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 3]
```diff
@@ -246,7 +247,7 @@ def add_move_id_field_account_bank_statement_line(env):
             account_bank_statement bs
         WHERE absl.statement_id = bs.id AND aj.company_id = bs.company_id
             AND absl.move_id IS NULL AND absl.payment_ref NOT IN ('', '/')
-            AND absl.payment_ref = ap.communication
+            AND absl.payment_ref = ap.communication AND ap.move_id IS NOT NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:250]: before 0:00:00.159084 running
after 0:00:00.171610 running

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 4]
```diff
@@ -306,6 +307,7 @@ def add_move_id_field_account_payment(env):
         SET payment_id = aml.payment_id
         FROM account_move_line aml
         WHERE aml.move_id = am.id AND aml.payment_id IS NOT NULL
+            AND am.payment_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:310]: before 0:00:18.865084 running
after 0:00:22.235087 running

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 5]
```diff
@@ -315,7 +317,7 @@ def add_move_id_field_account_payment(env):
         UPDATE account_payment ap
         SET move_id = am.id
         FROM account_move am
-        WHERE am.payment_id = ap.id""",
+        WHERE am.payment_id = ap.id AND ap.move_id IS NULL""",
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:320]: before 0:00:06.852229 running
after 0:00:03.768263 running

File: openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py [hunk 6]
```diff
@@ -349,7 +351,7 @@ def add_move_id_field_account_payment(env):
         UPDATE account_move am
         SET payment_id = ap.id
         FROM account_payment ap
-        WHERE am.id = ap.move_id
+        WHERE am.id = ap.move_id AND am.payment_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py:354]: before 0:00:06.550521 running
after 0:00:00.922105 running

File: openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py [hunk 7]
```diff
@@ -92,6 +92,7 @@ def fill_payment_id_and_statement_line_id_fields(env):
         SET payment_id = am.payment_id
         FROM account_move am
         WHERE am.id = aml.move_id AND am.payment_id IS NOT NULL
+            AND aml.payment_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:95]: before 0:00:48.115075 running
after 0:00:08.240818 running

File: openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py [hunk 8]
```diff
@@ -101,6 +102,7 @@ def fill_payment_id_and_statement_line_id_fields(env):
         SET statement_line_id = am.statement_line_id
         FROM account_move am
         WHERE am.id = aml.move_id AND am.statement_line_id IS NOT NULL
+            AND aml.statement_line_id IS NULL
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:105]: before 0:00:47.903751 running
after 0:00:01.759712 running

**Review by MiquelRForgeFlow — COMMENTED** [review]
Benchmarks in a client database migrated from v11:

- 64.342 records in account.invoice
- 222.854 records in account.invoice.line
- 198.276 records in account.move
- 732.860 records in account.move.line
- 42.899 records in account.payment
- 3.918 records in account.bank.statement
- 43.178 records in account.bank.statement.line

File: openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py (hunk 7 above)
Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account/14.0.1.1/post-migration.py:95]: This one seems the best improvement. Anyways, it doesn't seem too much gain in general, as this doesn't take a lot, but any improvement is welcome.

**Review by pedrobaeza — APPROVED** [review]
Thanks for the checking

**Pull request merged**